### PR TITLE
Interactives Index page tags plus some refactoring

### PR DESCRIPTION
**Pull request by geoffmrb**
### What

added the index page tags and also refactored some of the classes put in in the previous iteration

image

### How to review

Access interactives index page and add interactives or view existing, it will show tags next to it

### Who can review

i worked on the changes, anyone in the frontend team can review :)

### CODE REVIEW

File: src/app/views/interactives/InteractivesIndex.test.js [hunk 1]
```diff
@@ -22,8 +53,8 @@ describe("Collections", () => {
             expect(screen.getByLabelText("Internal ID")).toBeInTheDocument();
             expect(screen.getByLabelText("Title")).toBeInTheDocument();
             expect(screen.getByText("Interactive type")).toBeInTheDocument();
-            expect(screen.getByLabelText("Primary topic")).toBeInTheDocument();
-            expect(screen.getByLabelText("Data source")).toBeInTheDocument();
+            // expect(screen.getByLabelText("Primary topic")).toBeInTheDocument();
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesIndex.test.js:56]: this test has commented content, what is it testing?

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 2]
```diff
@@ -1,133 +1,86 @@
...
+                            Name - <b>{title}</b>
+                        </li>
+                        <li className="list-simple__item">
+                            Published date - <b>11 March 2022</b>
+                        </li>
+                        <li className="list-simple__item">
+                            Topic - <b>Health and social care (COVID-19)</b>
+                        </li>
+                    </ul>
+                    <p className="">
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:74]: remove empty className

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 3]
```diff
@@ -1,133 +1,86 @@
...
+                        Are you sure you want to delete this interactive? You will permanently lose access to the data associated to it, including the
+                        uploaded file.
+                    </p>
+                    <div className={"inline-block padding-top--2"}>
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:78]:
```suggestion
                    <div className="inline-block padding-top--2">
```

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 4]
```diff
@@ -1,133 +1,86 @@
...
-                            <ButtonWithShadow type="button" buttonText="Continue" onClick={this.handleDelete} isSubmitting={false} />
-                            <ButtonWithShadow type="button" class="secondary" buttonText="Cancel" onClick={this.handleReturn} isSubmitting={false} />
-                        </div>
+    return (
+        <FooterAndHeaderLayout title="Manage my interactives">
+            <div className="grid grid--justify-space-around padding-bottom--2">
+                <div className={"grid__col-8"}>
+                    <BackButton redirectUrl={`${rootPath}/interactives`} classNames={"ons-breadcrumb__item"} />
+                    <h1 className="text-align-left">Delete interactive</h1>
+                    <p className={"padding-bottom--1"}>You are about to delete this interactive:</p>
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:62]:
```suggestion
                    <p className="padding-bottom--1">You are about to delete this interactive:</p>
```

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 5]
```diff
@@ -1,133 +1,86 @@
...
-                        </p>
-                        <div className={"inline-block padding-top--2"}>
-                            <ButtonWithShadow type="button" buttonText="Continue" onClick={this.handleDelete} isSubmitting={false} />
-                            <ButtonWithShadow type="button" class="secondary" buttonText="Cancel" onClick={this.handleReturn} isSubmitting={false} />
-                        </div>
+    return (
+        <FooterAndHeaderLayout title="Manage my interactives">
+            <div className="grid grid--justify-space-around padding-bottom--2">
+                <div className={"grid__col-8"}>
+                    <BackButton redirectUrl={`${rootPath}/interactives`} classNames={"ons-breadcrumb__item"} />
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:60]:
```suggestion
                    <BackButton redirectUrl={`${rootPath}/interactives`} classNames="ons-breadcrumb__item" />
```

File: src/app/views/interactives/InteractivesIndex.test.js [hunk 6]
```diff
@@ -115,11 +151,12 @@ describe("Collections", () => {
         });
 
         it("should filter results when clicks apply button", () => {
+            useDispatchMock.mockReturnValue(filterInteractives);
             render(<InteractivesIndex {...defaultProps} />);
             const applyButton = screen.getByText("Apply");
             screen.getByLabelText("Internal ID").value = "Query";
             fireEvent.click(applyButton);
-            expect(defaultProps.filterInteractives.mock.calls).toHaveLength(1);
+            expect(filterInteractives).toHaveBeenCalled();
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesIndex.test.js:159]: can you assert the parameters here as well ?

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 7]
```diff
@@ -1,133 +1,86 @@
...
-            this.state.slug = metadata.slug;
-            this.state.published = metadata.published;
+    useEffect(() => {
+        if (successMessage.success) {
+            if (successMessage.type === "delete") {
+                props.router.push(`${rootPath}/interactives`);
+            }
         }
-    }
+    }, [successMessage]);
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:44]: it looks like your dependency is only [successMessage.success] here

File: src/app/views/interactives/InteractivesDelete.jsx [hunk 8]
```diff
@@ -1,133 +1,86 @@
...
+    useEffect(() => {
+        if (interactive.metadata) {
+            const { metadata } = interactive;
+            setInternalId(metadata.internal_id);
+            setTitle(metadata.title);
+            setLabel(metadata.label);
+            setSlug(metadata.slug);
+            setPublished(metadata.published);
         }
+    }, [interactive]);
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesDelete.jsx:36]:
```suggestion
    }, [interactive.metadata]);
```

File: src/app/views/interactives/InteractivesIndex.test.js (hunk 1 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesIndex.test.js:56]: a new filters by type feature but the backend isn't ready for it yet so commented it out for now...

File: src/app/views/interactives/InteractivesDelete.jsx (hunk 2 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesDelete.jsx]: removed

File: src/app/views/interactives/InteractivesDelete.jsx (hunk 3 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesDelete.jsx]: done

File: src/app/views/interactives/InteractivesDelete.jsx (hunk 4 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesDelete.jsx]: done

File: src/app/views/interactives/InteractivesIndex.test.js (hunk 6 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesIndex.test.js:232]: done

File: src/app/views/interactives/InteractivesDelete.jsx (hunk 7 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesDelete.jsx]: sorted

File: src/app/views/interactives/InteractivesIndex.jsx [hunk 9]
```diff
@@ -170,7 +170,7 @@ export default function InteractivesIndex(props) {
                                         <li key={key} className="list__item" role="listitem">
                                             <Link to={`${rootPath}/interactives/edit/${id}`} className="font-weight--600 font-size--18">
                                                 {metadata.label}
-                                            </Link>
+                                            </Link>{" "}
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesIndex.jsx:173]: remove this?

File: src/app/views/interactives/InteractivesIndex.test.js [hunk 10]
```diff
@@ -142,21 +143,144 @@ describe("Interactives index", () => {
...
+            rerender(<InteractivesIndex {...defaultProps} />);
+
+            const newItems = queryAllByRole("listitem");
+            expect(newItems.length).toBe(3);
+
             const applyButton = screen.getByText("Apply");
-            screen.getByLabelText("Internal ID").value = "Query";
+            screen.getByLabelText("Title").value = "Label 1";
             fireEvent.click(applyButton);
             expect(filterInteractives).toHaveBeenCalled();
```
Comment (AnnaEmson) [inline src/app/views/interactives/InteractivesIndex.test.js:232]: I meant to expect(filterInteractives).toHaveBeenCalledWith({...... });

File: src/app/views/interactives/InteractivesIndex.jsx (hunk 9 above)
Comment (AnnaEmson) [reply src/app/views/interactives/InteractivesIndex.jsx:173]: so can you please use `&nbsp;` instead

Comment (AnnaEmson) [reply src/app/views/interactives/InteractivesIndex.jsx:173]: or apply styling

File: src/app/views/interactives/InteractivesIndex.test.js (hunk 10 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesIndex.test.js]: addressed

File: src/app/views/interactives/InteractivesIndex.test.js [hunk 11]
```diff
@@ -142,21 +143,144 @@ describe("Interactives index", () => {
...
+            useSelectorMock.mockReturnValue(updatedState);
+
+            rerender(<InteractivesIndex {...defaultProps} />);
+
+            const newItems = queryAllByRole("listitem");
+            expect(newItems.length).toBe(3);
+
             const applyButton = screen.getByText("Apply");
-            screen.getByLabelText("Internal ID").value = "Query";
+            screen.getByLabelText("Title").value = "Label 1";
```
[the remark this answers is not in the document]
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesIndex.test.js]: addressed

File: src/app/views/interactives/InteractivesIndex.jsx (hunk 9 above)
Comment (geoffmrb) [reply src/app/views/interactives/InteractivesIndex.jsx]: done

**Review by AnnaEmson — APPROVED** [review]
nice work

**Pull request merged**